### DISCUSSION TITLE
「分享」在 Fly.io 平台上搭建 memos 并自动备份到 B2/S3

### DISCUSSION THREAD

**Discussion by hu3rror** [usememos/memos#451] [Ideas]
更新：新项目: [hu3rror/memos-litestream](https://github.com/hu3rror/memos-litestream) 欢迎来玩，可以在本地搭建带有 Litestream 功能的 memos~

---

大家好，我糊了一个在 Fly.io 上搭建 memos 的玩具 (memos-on-fly (https://github.com/hu3rror/memos-on-fly) ，默认重新 build 了 docker image，有兴趣可以玩玩。

这个玩具加了 [Litestream](https://litestream.io/) ，默认会把 memos 的 memos_prod.db 数据库自动备份到云存储桶 (bucket)，理论上支持 Backblaze B2，amazon S3，更多支持参考: https://litestream.io/guides/ 。

如果 fly.io 上的数据库被删除或意外丢失，就会自动从云存储桶重新下载到本地 (多一层保险)。哪怕 fly.io 倒闭，你都随时可以在 backblaze B2，amazon S3 手动把 memos_prod.db 下载回来，自己再本地搭建 memos，导入数据库就可以还原了。

想自己改 Dockerfile 和 相关脚本的可以 git clone 后切换到 build 分支。后续自己可以 git rebase --onto 或 git merge 同步 usememos/memos 官方的更新。

P.S. fly.io 信用卡验证后，可以免费开 3 个容器，内存 256MB/个，合计硬盘3GB，亚洲地区 30 GB 免费流量。没有信用卡的可以直接跳过本项目了😂

**Comment by realreus319** [comment]
请问备份的业务逻辑是什么，间隔时间？还是动态备份？

**Reply by hu3rror** [reply]
https://litestream.io/how-it-works/
